src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -818,12 +818,28 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
             {
                 return element switch
                 {
-                    BoundCollectionElementInitializer collectionInitializer => collectionInitializer.Arguments[collectionInitializer.InvokedAsExtensionMethod ? 1 : 0],
+                    BoundCollectionElementInitializer collectionInitializer => getCollectionInitializerElement(collectionInitializer),
                     BoundDynamicCollectionElementInitializer dynamicInitializer => dynamicInitializer.Arguments[0],
                     _ => null,
                 };
             }
             return element;
+
+            static BoundExpression getCollectionInitializerElement(BoundCollectionElementInitializer collectionInitializer)
+            {
+                int argIndex = collectionInitializer.InvokedAsExtensionMethod ? 1 : 0;
+                var arg = collectionInitializer.Arguments[argIndex];
+                Debug.Assert(!collectionInitializer.DefaultArguments[argIndex]);
+                if (collectionInitializer.Expanded && argIndex == collectionInitializer.AddMethod.ParameterCount - 1)
```

Comment:
>argIndex == collectionInitializer.AddMethod.ParameterCount - 1 [](http://example.com/codeflow?start=54&length=62)

Are we testing scenario when ```collectionInitializer.Expanded``` is true, but this condition is is false?

Reply:
Yes, `Add_ParamsArray_04`.